parlai/mturk/core/socket_manager.py

```diff
@@ -322,8 +322,11 @@ def _send_packet(self, packet, connection_id, send_time):
         if packet.requires_ack:
             if packet.blocking:
                 # Put the packet right back into its place to prevent sending
-                # other packets
+                # other packets, then block
                 self._safe_put(connection_id, (send_time, packet))
+                t = time.time() + self.ACK_TIME[packet.type]
+                while time.time() < t and packet.status != Packet.STATUS_ACK:
+                    time.sleep(0.2)
```

Comment:
assuming this is an arbitrary amount of time?

Reply:
Yes! I should use my arbitrary time constant instead.